pontoon/base/models.py

```diff
@@ -3441,3 +3445,14 @@ def serialize(self):
             "content": self.content,
             "id": self.id,
         }
+
+    def save(self, *args, **kwargs):
+        """
+        Validate Comments before saving.
+        """
+        if (self.translation is not None and (not self.locale and not self.entity)) or (
+            self.translation is None and (self.locale and self.entity)
+        ):
```

Comment:
You don't need to use `None` for the `translation` either, so I suggest you unify and simplify the checks:
```python
if (self.translation and not self.locale and not self.entity) or (
    not self.translation and self.locale and self.entity
):
```